docs/getting-started/ui/index.rst

```diff
@@ -0,0 +1,201 @@
+these kinds of features in your app.
+
+    Note: we don't currently provide a version of oso that runs in the
+    browser. This guide covers how to query for information in the
+    backend that can be sent to your frontend service.
+
+Getting a user's allowed actions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're familiar with oso, you know that oso policies contain
+``allow`` rules that specify that an **actor** is allowed to take an
+**action** on a **resource.**
+
+In many cases, like when you are authorizing a specific request in the
+backend, you'll just want to know if a specific
+``(actor, action, resource)`` combination is allowed, e.g., is
+``alice@gmail.com`` allowed to ``"READ"`` ``Expense{id: 1}``? For these
+yes/no queries, we provide the ``Oso.is_allowed()`` method.
+
+But when you're deciding what to display to Alice in the frontend, you
+may want more information. It can be especially useful to know **all the
+actions** that Alice is allowed to take on the ``Expense{id: 1}``
+resource, not just whether or not she can read it. In this case, you can
+use the ``Oso.get_allowed_actions()`` method to get a list of Alice's
+allowed actions and return them to your frontend.
+
+Let's look at an example. Imagine we have a GitHub-like app, that gives
+users access to repositories. The page to view a specific repository
+looks like this:
+
+.. image:: /getting-started/ui/a.png
+
+On this page there are several components that we may want to control
+based on the actions the current user is allowed to take. For example,
+we may want to hide the "Manage Access" or "Delete Repository" links
+depending on whether the user is allowed to take those actions.
+
+In this situation, the ``Oso.get_allowed_actions()`` method can be very
+helpful. The method returns a list of actions that a user is allowed to
+take on a specific resource. In our example, we call
+``get_allowed_actions()`` in the route handler for the "Show repository"
+view to get the user's allowed actions for the current repo:
+
+.. code:: python
+
+    def repos_show(org_id, repo_id):
+        # Get repo
+        repo = Repository.query.get(repo_id)
+        
+        # Authorize repo access
+        current_app.oso.authorize(repo, action="READ")
+
+        # Get allowed actions on the repo
+        actions = current_app.base_oso.get_allowed_actions(
+            get_current_user(), repo
+        )
+            # Send allowed actions to template (or frontend)
+        return render_template(
+            "repos/show.html",
+            repo=repo,
+            org_id=org_id,
+            actions=actions,
+        )
+
+In our demo app, when we call ``get_allowed_actions`` with the user
+``mike@monsters.com``, we get back:
+
+.. code:: python
+
+    actions = ['READ', 'LIST_ROLES', 'CREATE', 'DELETE', 'LIST_ISSUES']
+
+But when we call with a different user, ``sully@monsters.com``, we get:
+
+.. code:: python
+
+    actions = ['READ', 'LIST_ISSUES', 'CREATE']
```

Comment:
Is the ordering nondeterministic? Based on the returned actions for `mike@monsters.com` I would expect this list to be `['READ', 'CREATE', 'LIST_ISSUES']`.

Reply:
Even if it is (which it might be) I'll make it match this so it reads better.